test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -3937,9 +3937,45 @@ public async Task CustomISerializerSettingsFactory()
                 var status = client.GetStatusAsync();
 
                 Assert.NotNull(status);
+                Assert.Contains("Value2", status.Result.Output.ToString());
+            }
+        }
+
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task DefaultIMessageSerializerSettingsFactory()
+        {
+            string[] orchestratorFunctionNames =
+            {
+                nameof(TestOrchestrations.ComplexTypeOrchestrator),
+            };
 
-                var expectedResult = "Value2";
-                Assert.Contains(expectedResult, status.Result.Output.ToString());
+            using (var host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.DefaultIMessageSerializerSettingsFactory),
+                true,
+                serializerSettings: new DefaultEnumSettings()))

```

Comment:
I think this should be `null` instead of using a new settings object. If we accidentally changed the default, then this test would not catch it since default settings are duplicated in `DefaultEnumSettings`.

Reply:
"Default..." and "Custom..." are meant to be opposites of each other, so we dont have to rely on the default settings to test for custom settings. My "Default..." test doesn't actually use the default settings, it specifies its own. It just happens to match the current default settings.

Reply:
> My "Default..." test doesn't actually use the default settings, it specifies its own. It just happens to match the current default settings.

Right, this is exactly my concern. I don't think there is value in testing something that _matches_ the default settings. Instead I think we want to test the real default settings.

Reply:
I think there is value in having more than one test that tests different/opposite custom settings to be sure that the default settings didn't change to the same settings in my custom test. But I can add a real default test as well.